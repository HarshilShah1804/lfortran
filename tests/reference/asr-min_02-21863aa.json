{
    "basename": "asr-min_02-21863aa",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/min_02.f90",
    "infile_hash": "b683622b846877f83e88b48c637b382d1a967a1d5bf15622d056ab8e",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-min_02-21863aa.stdout",
    "stdout_hash": "dac1040b6140ff2921906b8fab9931fad237f2b3fa94cea3b656ae8c",
    "stderr": "asr-min_02-21863aa.stderr",
    "stderr_hash": "af77179f6e6513e1312ab2c0128d0e5c62b130b7e05dd68748884d6b",
    "returncode": 0
}