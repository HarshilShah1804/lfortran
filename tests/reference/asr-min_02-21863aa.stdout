(TranslationUnit
    (SymbolTable
        1
        {
            min_02:
                (Program
                    (SymbolTable
                        2
                        {
                            y:
                                (Variable
                                    2
                                    y
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            z:
                                (Variable
                                    2
                                    z
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    min_02
                    []
                    [(Print
                        (StringFormat
                            ()
                            [(IntrinsicElementalFunction
                                Min
                                [(Cast
                                    (Var 2 y)
                                    IntegerToInteger
                                    (Integer 8)
                                    ()
                                )
                                (Cast
                                    (Var 2 z)
                                    IntegerToInteger
                                    (Integer 8)
                                    ()
                                )]
                                0
                                (Integer 8)
                                ()
                            )]
                            FormatFortran
                            (String -1 0 () PointerString)
                            ()
                        )
                    )
                    (Print
                        (StringFormat
                            ()
                            [(StringConstant
                                "kind of min(y,z):"
                                (String 1 17 () PointerString)
                            )
                            (TypeInquiry
                                Kind
                                (Integer 8)
                                (IntrinsicElementalFunction
                                    Min
                                    [(Cast
                                        (Var 2 y)
                                        IntegerToInteger
                                        (Integer 8)
                                        ()
                                    )
                                    (Cast
                                        (Var 2 z)
                                        IntegerToInteger
                                        (Integer 8)
                                        ()
                                    )]
                                    0
                                    (Integer 8)
                                    ()
                                )
                                (Integer 4)
                                (IntegerConstant 8 (Integer 4) Decimal)
                            )]
                            FormatFortran
                            (String -1 0 () PointerString)
                            ()
                        )
                    )
                    (If
                        (IntegerCompare
                            (TypeInquiry
                                Kind
                                (Integer 8)
                                (IntrinsicElementalFunction
                                    Min
                                    [(Cast
                                        (Var 2 y)
                                        IntegerToInteger
                                        (Integer 8)
                                        ()
                                    )
                                    (Cast
                                        (Var 2 z)
                                        IntegerToInteger
                                        (Integer 8)
                                        ()
                                    )]
                                    0
                                    (Integer 8)
                                    ()
                                )
                                (Integer 4)
                                (IntegerConstant 8 (Integer 4) Decimal)
                            )
                            NotEq
                            (IntegerConstant 8 (Integer 4) Decimal)
                            (Logical 4)
                            (LogicalConstant
                                .false.
                                (Logical 4)
                            )
                        )
                        [(ErrorStop
                            (StringConstant
                                "Incorrect kind for min"
                                (String 1 22 () PointerString)
                            )
                        )]
                        []
                    )]
                )
        })
    []
)
